src/jit/morph.cpp

```diff
@@ -9550,8 +9550,7 @@ GenTreePtr          Compiler::fgMorphSmpOp(GenTreePtr tree, MorphAddrContext* ma
         {
             GenTreePtr effOp1 = op1->gtEffectiveVal();
             
-            noway_assert( (effOp1->gtOper == GT_CNS_INT) &&
-                         ((effOp1->gtIntCon.gtIconVal == 0) || (effOp1->gtIntCon.gtIconVal == 1)) );
+            noway_assert(effOp1->IsZero() || effOp1->IsIOne());
```

Comment:
I think that you actually mean that the condition is more liberal. However, by definition op1 will be of TYP_INT. So, although this adds (minor) extra liberality to the check, it keeps the code simple.